src/GC/GC.jl

```diff
@@ -3,77 +3,135 @@
 
 Garbage collection of Python objects.
 
-See `disable` and `enable`.
+See [`enable`](@ref), [`disable`](@ref) and [`gc`](@ref).
 """
 module GC
 
 using ..C: C
 
-const ENABLED = Ref(true)
 const QUEUE = C.PyPtr[]
+const QUEUE_LOCK = Threads.SpinLock()
```

Comment:
Nitpick is that I would prefer a pattern where these are one object so that it's more obvious to another developer that there is a lock associated to this queue, like:

```julia
const QUEUE = (;
    x=C.PyPtr[],
    lock=Threads.SpinLock()
)
```

and then you access it with

```julia
lock(QUEUE.lock) do
    f(QUEUE.x)
end
```

To me this is a more obvious association than the name + `_LOCK`. If a future dev does a global rename of `QUEUE`, they might forget to update the associated lock's name, and they would dissociate.